pkg/operation/botanist/secrets.go

```diff
@@ -159,18 +159,29 @@ func (b *Botanist) restoreSecretsFromShootStateForSecretsManagerAdoption(ctx con
 }
 
 func (b *Botanist) generateCertificateAuthorities(ctx context.Context) error {
+	generateOptions := []secretsmanager.GenerateOption{
+		secretsmanager.Persist(),
+		secretsmanager.Rotate(secretsmanager.KeepOld),
+	}
+
 	// TODO(rfranzke): Move this client CA secret configuration to the b.wantedCertificateAuthorities() function in a
 	//  future release.
 	if _, err := b.SecretsManager.Generate(ctx, &secretutils.CertificateSecretConfig{
 		Name:       v1beta1constants.SecretNameCAClient,
 		CommonName: "client",
 		CertType:   secretutils.CACert,
-	}, secretsmanager.Persist(), secretsmanager.Rotate(secretsmanager.KeepOld)); err != nil {
+	}, generateOptions...); err != nil {
 		return err
 	}
 
+	if shootStatus := b.Shoot.GetInfo().Status; shootStatus.Credentials == nil ||
+		shootStatus.Credentials.Rotation == nil ||
+		shootStatus.Credentials.Rotation.CertificateAuthorities == nil {
+		generateOptions = append(generateOptions, secretsmanager.IgnoreConfigChecksumForCASecretName())
```

Comment:
As soon, as we add the checksum to the CA secret names, we have to be careful, as changing configs will basically force rotate CAs of all existing cluster, correct?
But I assume, this is what we want (we will probably not change CA configs anyways).

Reply:
Yeah, this is indeed what we want for now. I also don't expect any CA config change soon (the current config is stable since project inception). If we need it we can still add some special logic for CA secrets.